02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,585 @@
+
+Now that you have added the styles and a header and footer to your application, you will proceed to set up authentication for requests to the Squidex API.
+
+## Step 5 — Preparing to fetch data from the Squidex API
+
+To be able to fetch data from your Squidex App, you need to have a bearer token. A token gives you the permission to create, read or modify content on a Squidex application. But before you can get a token, you need to create a client on Squidex.
+
+Open the dashboard of your blog application(`squidex-blog` or whatever name you chose) from https://cloud.squidex.io. On the left sidebar, click the **cog icon** at the bottom left corner of the page to open settings. Under the **Security** section click **Clients**. At the top of the page, you will see a section where you can create a client. In the text box provided under **Add a new client**, type `nextjs-blog` as the name of the client and click **Add Client** to create the client. Now when you scroll down the page you will notice a new `nextjs-blog` client has been created. By default, this new client comes with the Editor role. In Squidex, roles define which schemas a client has access to and which operations such as reading and writing a client is authorized to do. Since you wouldn't be making any changes to your content from your blog's frontend, you will change the role of the client to Reader i.e. with only read permissions. In the dropdown for **Role**, select **Reader**.
+
+Now, to get the token, click the **Connect** button for the `nextjs-blog` client. From the list of connection methods provided, select **Connect manually**. A token will be generated and will be displayed below the text reading **Just use the following token**. Click the copy icon to copy the generated token.
+
+After copying this token with Read access to your application, you will create a file containing environment variables for your Next.js app. Create a `.env.local` file in the `squidex-blog` folder and add the following contents
+
+```
+SQUIDEX_API_URL=https://cloud.squidex.io/api/content/squidex-blog/graphql
+SQUIDEX_BEARER_TOKEN=API-KEY
+```
+
+Replace `squidex-blog` in SQUIDEX_API_URL with the name of your app on Squidex and replace `API-KEY` with the token you copied from the dashboard.
+
+These environment variables will not be visible on the client-side bundle since they will only be used at build time to authenticate queries to the Squidex API. In Next.js, the only environment variables that are available on client-side are those prefixed with `NEXT_PUBLIC_`. You can learn more about environment variables in Next.js [here](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser).
+
+**NOTE:** When you deploy your app to production, make sure to replace the Squidex token every month since the tokens are valid for only a month.
```

Comment:
This makes no sense, there should be a solution to create a bearer token on the fly.

Reply:
How about we create a client with the _Reader_ role and enable anonymous access so the reader wouldn't need tokens to make requests?

Reply:
Yes, this works as well but a manual process would fail after a few months.

Reply:
I think this is the last important piece to be added.

Reply:
I had updated the article to use anonymous access instead.
Later, I would follow the steps in the tutorial from start till finish to make sure I didn't break the tutorial while making edits.